modules/services/muchsync.nix

```diff
@@ -0,0 +1,160 @@
+        If muchsync dies with the error "message received does not match hash,"
+        you likely need to run it with the -F option.
+
+        Note that if your software regularly modifies the contents of mail files
+        (e.g., because you are running offlineimap with "synclabels = yes"),
+        then you will need to use -F each time you run muchsync.
+        Specify it as a server option (after the server name) if the editing happens server-side.
+      '';
+    };
+
+    remoteBinaryPath = mkOption {
+      type = types.str;
+      default = "";
+      description = ''
+        Specifies the path to muchsync on the server.
+        Ordinarily, muchsync should be in the default PATH on the server so this option is not required.
+        However, this option is useful if you have to install muchsync in a non-standard place or wish
+        to test development versions of the code.
+      '';
+    };
+
+    sshCommand = mkOption {
+      type = types.str;
+      default = "${pkgs.openssh}/bin/ssh -CTaxq";
+      defaultText = "ssh -CTaxq";
+      description = ''
+        Specifies a command line to pass to /bin/sh to execute a command on another machine.
+        The default value is "ssh -CTaxq".
+        Note that because this string is passed to the shell,
+        special characters including spaces may need to be escaped.
+      '';
+    };
+
+    noNew = mkOption {
+      type = types.bool;
+      default = false;
+      description = ''
+        Ordinarily, muchsync begins by running "notmuch new".
+        This option says not to run "notmuch new" before starting the muchsync operation.
+        It can be passed as either a client or a server option.
+        For example: The command "muchsync myserver --nonew" will run "notmuch new" locally but not on myserver.
+      '';
+    };
+
+    noUpload = mkOption {
+      type = types.bool;
+      default = false;
+      description = ''
+        Transfer files from the server to the client, but not vice versa.
+      '';
+    };
+
+    backgroundUpload = mkOption {
+      type = types.bool;
+      default = false;
+      description = ''
+        Transfer files from the server to the client in the foreground.
+        Then fork into the background to upload any new files from the client to the server.
+        This option is useful when checking new mail, if you want to begin reading your mail
+        as soon as it has been downloaded while the upload continues.
+      '';
+    };
+  };
+
+  config = mkIf cfg.enable {
+    systemd.user.services.muchsync = {
+      Unit = {
+        Description = "muchsync sync service";
+        PartOf = [ "network-online.target" ];
+      };
+      Service = {
+        CPUSchedulingPolicy = "idle";
+        IOSchedulingClass = "idle";
+        Environment = [
+          ''"NOTMUCH_CONFIG=${config.home.sessionVariables.NOTMUCH_CONFIG}"''
+          ''"NMBGIT=${config.home.sessionVariables.NMBGIT}"''
```

Comment:
Since these are set by the notmuch module I think it would be nice to have an assertion that verifies that the notmuch module is enabled. Like in mpdris2:
https://github.com/rycee/home-manager/blob/d625186ce599a663b0115b83b70ea0be0ba8afbf/modules/services/mpdris2.nix#L79-L84

Reply:
Assertion added.